### PR TITLE
Add optional Standard settings support

### PR DESCRIPTION
**Pull request by PoignardAzur**

### CODE REVIEW

File: src/getSettings.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (emmanuelgeoffray) [inline src/getSettings.js]: overkill to have one file just for that, can you put it in `spacebro-client.js` ?

File: src/spacebro-client.js [hunk 2]
```diff
@@ -53,7 +74,13 @@ class SpacebroClient {
     this.socket = null
 
     if (address == null && port == null) {
-      return
+      if (getSettings() && spacebroSettings) {
+        address = spacebroSettings.host
```
Comment (emmanuelgeoffray) [inline src/spacebro-client.js]: messy. Let's talk about this IRL.

File: src/spacebro-client.js [hunk 3]
```diff
@@ -237,17 +264,39 @@ let spacebroClientSingleton = null
 /*
 ** This variable is used to allow calling `on` before `connect`
 */
-let beforeConnectSpacebroClient = new SpacebroClient()
+let beforeConnectEvents = {
```
Comment (emmanuelgeoffray) [inline src/spacebro-client.js]: what's the link with the branch's purpose?

File: test/tests-connect.js [hunk 4]
```diff
@@ -16,20 +16,6 @@ test('connect', async (t) => {
   await sleep(5000)
 })
 
-test('connect - Delayed', async (t) => {
```
Comment (emmanuelgeoffray) [inline test/tests-connect.js]: why was this removed?

File: test/tests-without-standard.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  _dontGetSettings()
+  /* eslint-disable no-new */
+  new SpacebroClient()
+
+  console.warn = consoleWarn
+  t.deepEqual(warnings, [
+    'SpacebroClient instance constructed without options; and service.spacebro not found in standard-settings',
+    'Cannot connect without host address and port from standard-settings'
+  ])
+})
```
Comment (emmanuelgeoffray) [inline test/tests-without-standard.js]: test a normal connect with options when standard-settings disabled

File: src/getSettings.js (hunk 1 above)
Comment (PoignardAzur) [inline src/getSettings.js]: Right

File: test/tests-connect.js (hunk 4 above)
Comment (PoignardAzur) [inline test/tests-connect.js]: Because the semantics for a constructor without parameters were changed

File: src/spacebro-client.js (hunk 3 above)
Comment (PoignardAzur) [inline src/spacebro-client.js]: `new SpacebroClient()` had side effects; I replaced it with an object because using a 'client' in that situation was a technical debt anyway.

File: test/tests-without-standard.js (hunk 5 above)
Comment (PoignardAzur) [inline test/tests-without-standard.js]: Done. In the next commit, all tests have standard-settings disabled by default.

File: src/spacebro-client.js (hunk 2 above)
Comment (PoignardAzur) [inline src/spacebro-client.js]: Ok. Code streamlined.
